File: Code/Backview/src/api/post.ts
/** 内容管理 API
 * 对接真实后台接口 */

import service from '@/utils/request'
import type { PageQuery } from '@/types'

// ===== 帖子审核 =====

// 待审核帖子列表
export function listPostReviews(query: PageQuery & {
  type?: string
  start_time?: string
  end_time?: string
}) {
  return service.get<{ list: PostReviewItem[]; total: number; page: number; size: number }>('/content/posts/pending', { params: query })
}

// 后端返回的帖子列表项结构
export interface PostReviewItem {
  id: string
  type: string
  title: string
  content: string
  desc?: string
  cover?: string
  media: MediaItem[]
  diagram_id?: string
  like_count: number
  comment_count: number
  favorite_count: number
  view_count: number
  review_status: string
  status: string
  risk_level?: 'none' | 'low' | 'mid' | 'high'
  risk_tags?: string[]
  create_time: string
  publish_time: string
  ip?: string
  device?: string
  topic_ids?: string[]
  topics?: string[]
  bead_params?: string
  author: {
    id: string
    nickname: string
    avatar?: string
    is_member?: boolean
  }
  is_liked: boolean
  is_favorited: boolean
}

export interface MediaItem {
  type: 'image' | 'video'
  url: string
  width?: number
  height?: number
}

// 帖子详情
export function getPost(id: string) {
  return service.get<PostDetailItem>('/content/posts/' + id)
}

export interface PostDetailItem extends PostReviewItem {
  topic_ids?: string[]
  bead_params?: string
}

// 审核通过帖子
export function approvePost(id: string) {
  return service.post('/content/posts/' + id + '/approve')
}

// 驳回帖子
export function rejectPost(id: string, reason?: string) {
  return service.post('/content/posts/' + id + '/reject', { reason })
}

// 批量审核通过
export function batchApprovePosts(ids: string[]) {
  return service.post('/content/posts/batch-approve', { post_ids: ids })
}

// 批量驳回
export function batchRejectPosts(ids: string[], reason?: string) {
  return service.post('/content/posts/batch-reject', { post_ids: ids, reason })
}

// ===== 评论管理 =====

// 评论列表
export function listCommentReviews(query: PageQuery & {
  post_id?: string
  status?: string
}) {
  return service.get<{ list: CommentItem[]; total: number }>('/content/comments', { params: query })
}

export interface CommentItem {
  id: string
  post_id: string
  post_title: string
  user: {
    id: string
    nickname: string
    avatar?: string
  }
  content: string
  ip: string
  status: string
  create_time: string
}

// 隐藏评论
export function hideComment(id: string) {
  return service.post('/content/comments/' + id + '/hide')
}

// 审核通过评论（与隐藏相同逻辑）
export function approveComment(id: string) {
  return service.post('/content/comments/' + id + '/hide')
}

// 删除评论
export function deleteComment(id: string) {
  return service.delete('/content/comments/' + id)
}

// ===== 敏感词管理 =====

// 敏感词列表
export function listSensitiveWords(query: { page?: number; page_size?: number; level?: number; type?: string; keyword?: string } = {}) {
  return service.get<{ list: SensitiveWordItem[]; total: number }>('/content/sensitive-words', { params: query })
}

export interface SensitiveWordItem {
  id: string
  word: string
  level: number
  type: string
  replace_word?: string
  status: number
  create_time: string
}

// 添加敏感词
export function addSensitiveWord(data: { word: string; level: number; type: string; replace_word?: string }) {
  return service.post<{ id: string }>('/content/sensitive-words', data)
}

// 更新敏感词
export function updateSensitiveWord(id: string, data: { word?: string; level?: number; type?: string; replace_word?: string }) {
  return service.put('/content/sensitive-words/' + id, data)
}

// 删除敏感词
export function deleteSensitiveWord(id: string) {
  return service.delete('/content/sensitive-words/' + id)
}

// ===== 举报管理 =====

// 举报列表
export function listReports(query: PageQuery & {
  status?: string
  type?: string
}) {
  return service.get<{ list: ReportItem[]; total: number }>('/content/reports', { params: query })
}

export interface ReportItem {
  id: string
  report_id: string
  reporter: {
    id: string
    nickname: string
  }
  target_type: string
  target_id: string
  target_content?: string
  reason: string
  images?: string[]
  status: string
  create_time: string
}

// 处理举报
export function handleReport(id: string, action: string, result?: string) {
  return service.post('/content/reports/' + id + '/handle', { action, result })
}

File: Code/APP/lib/models/post.dart
class Post {
  final String id;
  final String userId;
  final String type;
  final String title;
  final String content;
  final List<PostMedia> media;
  final List<String> topicIds;
  final Map<String, dynamic>? beadParams;
  final String? diagramId;
  final int likeCount;
  final int commentCount;
  final int favoriteCount;
  final int shareCount;
  final String status;
  final String createdAt;
  final String publishedAt;
  final PostUser? user;

  Post({
    required this.id,
    required this.userId,
    required this.type,
    required this.title,
    required this.content,
    required this.media,
    required this.topicIds,
    this.beadParams,
    this.diagramId,
    this.likeCount = 0,
    this.commentCount = 0,
    this.favoriteCount = 0,
    this.shareCount = 0,
    required this.status,
    required this.createdAt,
    required this.publishedAt,
    this.user,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'] ?? '',
      userId: json['user_id'] ?? '',
      type: json['type'] ?? 'work',
      title: json['title'] ?? '',
      content: json['content'] ?? '',
      media: (json['media'] as List<dynamic>?)
          ?.map((m) => PostMedia.fromJson(m))
          .toList() ?? [],
      topicIds: (json['topic_ids'] as List<dynamic>?)
          ?.map((t) => t.toString())
          .toList() ?? [],
      beadParams: json['bead_params'],
      diagramId: json['diagram_id'],
      likeCount: json['like_count'] ?? 0,
      commentCount: json['comment_count'] ?? 0,
      favoriteCount: json['favorite_count'] ?? 0,
      shareCount: json['share_count'] ?? 0,
      status: json['status'] ?? 'published',
      createdAt: json['created_at'] ?? '',
      publishedAt: json['published_at'] ?? '',
      user: json['user'] != null ? PostUser.fromJson(json['user']) : null,
    );
  }
}

class PostMedia {
  final String url;
  final String type;
  final int width;
  final int height;

  PostMedia({
    required this.url,
    required this.type,
    required this.width,
    required this.height,
  });

  factory PostMedia.fromJson(Map<String, dynamic> json) {
    return PostMedia(
      url: json['url'] ?? '',
      type: json['type'] ?? 'image',
      width: json['width'] ?? 0,
      height: json['height'] ?? 0,
    );
  }
}

class PostUser {
  final String userId;
  final String nickname;
  final String avatar;
  final bool isFollowed;

  PostUser({
    required this.userId,
    required this.nickname,
    required this.avatar,
    this.isFollowed = false,
  });

  factory PostUser.fromJson(Map<String, dynamic> json) {
    return PostUser(
      userId: json['user_id'] ?? '',
      nickname: json['nickname'] ?? '',
      avatar: json['avatar'] ?? '',
      isFollowed: json['is_followed'] ?? false,
    );
  }
}

class Comment {
  final String id;
  final String postId;
  final String userId;
  final String content;
  final String? parentId;
  final String? replyToUserId;
  final int likeCount;
  final String createdAt;
  final CommentUser? user;

  Comment({
    required this.id,
    required this.postId,
    required this.userId,
    required this.content,
    this.parentId,
    this.replyToUserId,
    this.likeCount = 0,
    required this.createdAt,
    this.user,
  });

  factory Comment.fromJson(Map<String, dynamic> json) {
    return Comment(
      id: json['id'] ?? '',
      postId: json['post_id'] ?? '',
      userId: json['user_id'] ?? '',
      content: json['content'] ?? '',
      parentId: json['parent_id'],
      replyToUserId: json['reply_to_user_id'],
      likeCount: json['like_count'] ?? 0,
      createdAt: json['created_at'] ?? '',
      user: json['user'] != null ? CommentUser.fromJson(json['user']) : null,
    );
  }
}

class CommentUser {
  final String userId;
  final String nickname;
  final String avatar;

  CommentUser({
    required this.userId,
    required this.nickname,
    required this.avatar,
  });

  factory CommentUser.fromJson(Map<String, dynamic> json) {
    return CommentUser(
      userId: json['user_id'] ?? '',
      nickname: json['nickname'] ?? '',
      avatar: json['avatar'] ?? '',
    );
  }
}